packages/web/next-env.d.ts

```diff
@@ -1,3 +1,4 @@
 /// <reference types="next" />
 /// <reference types="next/types/global" />
 declare module '*.png';
+declare module '@metafam/ds';
```

Comment:
Shouldnt do this, we will lose type-safety. What was the reason you added this?

Reply:
@hammadj I'm actually getting type errors when I run `yarn typecheck` without this line:
```
components/RegisterPlayer.tsx(14,20): error TS2322: Type '{ children: Element[]; flex: number; }' is not assignable to type 'IntrinsicAttributes & ComponentWithAs<"div", FlexProps> & { children?: ReactNode; }'.
  Property 'flex' does not exist on type 'IntrinsicAttributes & ComponentWithAs<"div", FlexProps> & { children?: ReactNode; }'.
components/RegisterPlayer.tsx(17,25): error TS2322: Type '{ children: (string | Element)[]; textTransform: string; }' is not assignable to type 'IntrinsicAttributes & ComponentWithAs<"button", ButtonProps> & { children?: ReactNode; }'.
  Property 'textTransform' does not exist on type 'IntrinsicAttributes & ComponentWithAs<"button", ButtonProps> & { children?: ReactNode; }'.
components/RegisterPlayer.tsx(20,10): error TS2322: Type '{ children: string; }' is not assignable to type 'ComponentWithAs<"button", ButtonProps>'.
  Type '{ children: string; }' provides no match for the signature '<TT extends ElementType<any>>(props: PropsWithAs<TT, ButtonProps>): ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | null'.
components/RegisterPlayer.tsx(21,10): error TS2322: Type '{ children: (string | Element)[]; }' is not assignable to type 'ComponentWithAs<"button", ButtonProps>'.
  Type '{ children: (string | Element)[]; }' provides no match for the signature '<TT extends ElementType<any>>(props: PropsWithAs<TT, ButtonProps>): ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | null'.
components/SuccessPlayer.tsx(10,20): error TS2322: Type '{ children: Element[]; h: string; }' is not assignable to type 'IntrinsicAttributes & ComponentWithAs<"div", FlexProps> & { children?: ReactNode; }'.
  Property 'h' does not exist on type 'IntrinsicAttributes & ComponentWithAs<"div", FlexProps> & { children?: ReactNode; }'.
components/WelcomePlayer.tsx(13,20): error TS2322: Type '{ children: Element[]; flex: number; maxWidth: string; }' is not assignable to type 'IntrinsicAttributes & ComponentWithAs<"div", FlexProps> & { children?: ReactNode; }'.
  Property 'flex' does not exist on type 'IntrinsicAttributes & ComponentWithAs<"div", FlexProps> & { children?: ReactNode; }'.
pages/login.tsx(14,20): error TS2322: Type '{ children: (false | Element)[]; backgroundImage: string; }' is not assignable to type 'IntrinsicAttributes & ComponentWithAs<"div", FlexProps> & { children?: ReactNode; }'.
  Property 'backgroundImage' does not exist on type 'IntrinsicAttributes & ComponentWithAs<"div", FlexProps> & { children?: ReactNode; }'.
pages/profile.tsx(9,20): error TS2322: Type '{ children: Element; backgroundImage: string; }' is not assignable to type 'IntrinsicAttributes & ComponentWithAs<"div", FlexProps> & { children?: ReactNode; }'.
  Property 'backgroundImage' does not exist on type 'IntrinsicAttributes & ComponentWithAs<"div", FlexProps> & { children?: ReactNode; }'.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
Actually the issue was that I was using `typeof Button` instead of `React.ComponentProps<typeof Button>`. 
That fixed this issue!